### PR TITLE
Load Tilesets From External Files

### PR DESCRIPTION
**Pull request by guodman**
When the tileset is in an external file, parse that file when given the path of the map file.  To load an extenal tileset, the new parse_file function must be used.  If the map has an external tileset and the original parse endpoint is used, the result will be an error.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
Would you be opposed to publicly exporting TSX parsing, too?

I could use that in a project of mine.

**Comment by guodman** [comment]
That should be easy to do.  I'll work on that this weekend and update this pull request.

File: src/lib.rs [hunk 1]
```diff
@@ -351,7 +358,62 @@ impl Tileset {
...
+
+    fn parse_external_tileset<R: Read>(g: u32, parser: &mut EventReader<R>, attrs: &Vec<OwnedAttribute>) -> Result<Tileset, TiledError> {
+        let ((s, m), (n, w, h)) = get_attrs!(
+            attrs,
+            optionals: [("spacing", spacing, |v:String| v.parse().ok()),
+                        ("margin", margin, |v:String| v.parse().ok())],
+            required: [("name", name, |v| Some(v)),
+                       ("tilewidth", width, |v:String| v.parse().ok()),
+                       ("tileheight", height, |v:String| v.parse().ok())],
+            TiledError::MalformedAttributes("tileset must have a firstgid, name tile width and height with correct types".to_string()));
```
Comment (Phrohdoh) [inline src/lib.rs]: `firstgid, name, tile, width, and height` is easier to comprehend.

**Comment by guodman** [comment]
I have updated this PR with a public method for parsing just a tsx file, as well as the other feedback you gave.  Thanks for the suggestions.

**Comment by mattyhall** [comment]
Sorry for the delay. This is really cool, cheers!

**Pull request merged**